commit dd9ef5591da27eadcee4ebb3d69f0e43639bb21b (refs/changes/72/2772/4)
Author: Chris Burroughs <chris.burroughs@joyent.com>
Date:   2017-10-30T15:53:44+00:00 (1 year, 11 months ago)
    
    HEAD-2369 cache=false with sdc-server setup still creates a cache device
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/scripts/joysetup.sh b/scripts/joysetup.sh
index 97b83d34..2b4e6cfb 100755
--- a/scripts/joysetup.sh
+++ b/scripts/joysetup.sh
@@ -625,7 +625,8 @@ if [[ "$(zpool list)" == "no pools available" ]] \
         dl3="${CONFIG_layout}"
     fi
 
-    if ! /usr/bin/disklayout ${dl0} "${dl1}" ${dl2} "${dl3}"> ${POOL_FILE}; then
+    if ! /usr/bin/disklayout ${dl0} ${dl1:+"$dl1"} ${dl2} "${dl3}" \
+         > ${POOL_FILE}; then
         fatal "disk layout failed"
     fi
 
